lib/jxl/decode.cc

```diff
@@ -2185,4 +2201,6 @@ JxlDecoderStatus JxlDecoderSetPreferredColorProfile(
 
 void SetDecoderMemoryLimitBase_(size_t memory_limit_base) {
   memory_limit_base_ = memory_limit_base;
+  // Allow 5 x max_image_size processing units.
+  cpu_limit_base_ = 5 * memory_limit_base;
```

Comment:
Doesn't this result in a hard limit on the # of frames in an image?

Reply:
nope. it is sum(w * h) limit.

Reply:
Sure, but if you have a long animation you'll go over this number, no?

Reply:
Of course. 1x1 animation in this case is allowed to have no more than 10M frames.
But it does not limit the number of frames - limitation is applied to frames being decoded. IIUC skipped frames won't count for that limit.